autometrics/src/otel_push_exporter.rs

```diff
@@ -0,0 +1,131 @@
+use opentelemetry_api::metrics::MetricsError;
+use opentelemetry_otlp::{ExportConfig, Protocol, WithExportConfig};
+use opentelemetry_otlp::{OtlpMetricPipeline, OTEL_EXPORTER_OTLP_TIMEOUT_DEFAULT};
+use opentelemetry_sdk::metrics::MeterProvider;
+use opentelemetry_sdk::runtime;
+use std::ops::Deref;
+use std::time::Duration;
+
+#[repr(transparent)]
+#[must_use = "Assign this to a unused variable instead: `let _meter = ...` (NOT `let _ = ...`), as else it will be dropped immediately - which will cause it to be shut down"]
+pub struct OtelMeterProvider(MeterProvider);
+
+impl Deref for OtelMeterProvider {
+    type Target = MeterProvider;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Drop for OtelMeterProvider {
+    fn drop(&mut self) {
+        // this will only error if `.shutdown` gets called multiple times
+        let _ = self.0.shutdown();
+    }
+}
+
+#[cfg(feature = "otel-push-exporter-http")]
+pub fn init_http(url: impl Into<String>) -> Result<OtelMeterProvider, MetricsError> {
+    init_http_with_timeout_period(
+        url,
+        Duration::from_secs(OTEL_EXPORTER_OTLP_TIMEOUT_DEFAULT),
+        Duration::from_secs(1),
```

Comment:
This `period` is that the same as push interval? If so, then I think we should not use such a low value, not sure what would be a better value :thinking: Maybe something like 30 seconds.

Reply:
im actually thinking maybe we shouldnt set these at all. then otlp exporter would fall back to using the env variables `OTEL_METRIC_EXPORT_TIMEOUT` and `OTEL_METRIC_EXPORT_INTERVAL` and if not found fall back to the default value, currently `30` timeout and `60` interval

Reply:
i have done so now

Reply:
I forgot about this, using 30 seconds is probably the better default, but also maybe it'll create struggles with the explorer with the build_info metric being looked for only on the last second. Didn't have time to test Brett's fix, maybe that's not an issue anymore